app/views/posts/_list_item.haml

```diff
@@ -11,9 +11,14 @@
     - if !current_user.try(:hide_warnings) && post.content_warnings.exists?
       = image_tag "/images/exclamation.png", class: 'vmid', title: "Content Warning: " + post.content_warnings.pluck(:name).join(', '), alt: '!'
   %td.post-subject{class: klass}
-    - if logged_in? && (@show_unread || (!@unread_ids.nil? && @unread_ids.include?(post.id)))
+    - if logged_in? && ((!@opened_ids.nil? && !@opened_ids.include?(post.id)) || (!@unread_ids.nil? && @unread_ids.include?(post.id)))
```

Comment:
(should probably clarify that these are my personal gripes with things like "line is too long" and "too many conditions" and "lots of parentheses", my existing style set no good precedent here, this is not your fault I'm just griping to gripe)